stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2473,6 +2473,11 @@ collections = [
 /// keys, followed by the values.
 //
 // See the docs at the top of the file for more details on this type
+//
+// NOTE: The precise layout of this type is relied on in the runtime
+// to provide a statically allocated empty singleton. 
```

Comment:
On the plus side, everything in these types is literally pointers, pointer-sized integers, and composites thereof, so there's a lot of resilient layout strategies that won't randomly break it (there's no benefit to be had in reordering fields).